model_analyzer/config/generate/quick_run_config_generator.py

```diff
@@ -404,17 +407,72 @@ def _get_next_perf_analyzer_config(self, model_variant_name: str,
         perf_analyzer_config.update_config_from_profile_config(
             model_variant_name, self._config)
 
-        model_batch_size = dimension_values.get("max_batch_size", 1)
-        instance_count = dimension_values.get("instance_count", 1)
+        dimension_batch_size = int(dimension_values.get("max_batch_size", 1))
+        dimension_instance_count = int(dimension_values.get(
+            "instance_count", 1))
 
-        concurrency = 2 * model_batch_size * instance_count
+        model_batch_size = self._calculate_model_batch_size(
+            dimension_batch_size)
+        instance_count = self._calculate_instance_count(
+            dimension_instance_count)
+
+        concurrency = self._calculate_concurrency(model_batch_size,
+                                                  instance_count)
 
         perf_config_params = {'batch-size': 1, 'concurrency-range': concurrency}
         perf_analyzer_config.update_config(perf_config_params)
 
         perf_analyzer_config.update_config(model.perf_analyzer_flags())
         return perf_analyzer_config
 
+    def _calculate_model_batch_size(self, batch_size: int) -> int:
+        min_batch_size_set = self._config.get_config(
+        )['run_config_search_min_model_batch_size'].set_by_config()
+
+        max_batch_size_set = self._config.get_config(
+        )['run_config_search_max_model_batch_size'].set_by_config()
+
+        if min_batch_size_set and batch_size < self._config.run_config_search_min_model_batch_size:
```

Comment:
My thought was that we would constrain the dimensions. It seems your approach was to make the dimensions above a certain point clamped to the min/max value. Does that have any implication on the algorithm or the ability to rerun?

Reply:
You bring up a good point, It doesn't have any implication on the algorithm, but it does present a rerun issue. As the checkpoint result is not valid if the user specifies a restriction, we clamp, and then reruns without the restriction. On the subsequent run we will use the clamped result.

I'll need to redo this to clamp the dimension in the algorithm.

Reply:
It doesn't effect rerun, since we're modifying the model/PA config, which is stored in the checkpoint. So, if the user reruns without clamping we will create a new config.

I'm not convinced this negatively effects the algorithm, as I don't think it cares about the clamping, but we should talk.

Reply:
I'm good with what we have. I've also added some documentation. Please re-review.